File: Source/Fantasy.Jobs/Solar/ISatellite.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using Fantasy.Jobs.Management;
using Fantasy.Jobs.Resources;
using System.ServiceModel;

namespace Fantasy.Jobs.Solar
{
    [ServiceContract(Namespace=Consts.JobServiceNamespaceURI)]
    public interface ISatellite
    {
        [OperationContract]
        void Echo();

        

        #region Job queue events

        [OperationContract]
        void OnJobAdded(JobMetaData job);

        [OperationContract]
        void OnJobChanged(JobMetaData job);

        #endregion

        #region Resource service

        [OperationContract]
        bool IsResourceAvailable(ResourceParameter[] parameters);


        [OperationContract]
        double GetLoadFactor();

        #endregion

        #region Job Controller

        [OperationContract]
        void RequestStartJob(JobMetaData job);

        [OperationContract]
        void RequestResume(JobMetaData job);

        [OperationContract]
        void RequestCancel(Guid id);

        [OperationContract]
        void RequestSuspend(Guid id);

        [OperationContract]
        void RequestUserPause(Guid id);

        [OperationContract]
        void RequestSuspendAll();


        [OperationContract]
        bool IsRunning(Guid id);

        #endregion 

        event EventHandler<JobQueueEventArgs> JobChanged;

        event EventHandler<JobQueueEventArgs> JobAdded;

        
    }
}

File: Fantasy.Jobs/src/fantasy/jobs/solar/ISatellite.java
package fantasy.jobs.solar;

import java.rmi.Remote;
import java.util.UUID;

import fantasy.jobs.management.*;
import fantasy.jobs.resources.*;


public interface ISatellite extends Remote
{

	void echo() throws Exception;

	void onJobAdded(JobMetaData job) throws Exception;

	void onJobChanged(JobMetaData job) throws Exception;


	boolean isResourceAvailable(ResourceParameter[] parameters) throws Exception;

	double getLoadFactor()  throws Exception;

	void requestStartJob(JobMetaData job)  throws Exception;


	void requestResume(JobMetaData job) throws Exception;


	void requestCancel(UUID id) throws Exception;

    void requestSuspend(UUID id) throws Exception;

	void requestUserPause(UUID id) throws Exception;

	void requestSuspendAll() throws Exception;


	boolean isRunning(UUID id) throws Exception;


	
	void addListener(ISatelliteListener listener);
	
	void removeListener(ISatelliteListener listener);

	String getName();

}